### PR TITLE
http_err.et: add missing 510 (Not Extended)

### PR DESCRIPTION
**Pull request by ksmurchison**
No real reason for this other than I stumbled across it and figured I'd add it for completeness

### CODE REVIEW

File: imap/http_err.et [hunk 1]
```diff
@@ -247,6 +248,9 @@ ec HTTP_NO_STORAGE,
 ec HTTP_LOOP,
    "508 Loop Detected"                   /* RFC 5842 (Binding Extn to WebDAV) */
 
+ec HTTP_LOOP,
+   "510 Not Extended"                    /* RFC 2774 (An HTTP Extn Framework) */
```
[the remark this answers is not in the document]
Comment (ksmurchison) [reply imap/http_err.et:252]: Fixed.  Thanks.

**Comment by dilyanpalauzov** [comment]
Cyrus IMAP is full of code, which does nothing, but is nevertheless there.  Apart from making the maintenance more difficult, the unused code complicates the learning curve for new developers.  As this HTTP-code is not needed right now, I am against increasing the code base with it.

**Review by rsto — COMMENTED** [review]
I agree with @dilyanpalauzov that unused code makes the code base unnecessarily hard to learn. But IMHO this does not apply to standard HTTP error codes. In any case, I won't object either decision.

**Comment by ksmurchison** [comment]
I think having the complete list of HTTP response codes is useful, but I've commented out the ones we don't currently use to avoid bloat in the generated array.

**Review by rsto — APPROVED** [review]

**Pull request merged**